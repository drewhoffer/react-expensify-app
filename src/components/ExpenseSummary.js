import React from 'react';
import { connect } from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';


 export const ExpenseSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
     return (
        <div>
        {
            expensesCount === 0 ? <h1></h1> :
             <h1>
                Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}
            </h1>
        }
            
        
        </div>
     );
};

const mapStateToProps = (state) => {
    const filteredExpenses = (selectExpenses(state.expenses, state.filters));
    return {
        expensesCount: filteredExpenses.length,
        expensesTotal: expensesTotal(filteredExpenses)
    };
};


export default connect(mapStateToProps)(ExpenseSummary);
