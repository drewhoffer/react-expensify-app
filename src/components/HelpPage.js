import React from 'react';

const HelpPage = () => (
    <div>
        Some Help page
    </div>
);

export default HelpPage;