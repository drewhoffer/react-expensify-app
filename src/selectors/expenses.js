//get visible expenses
import moment from 'moment';



export default (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter((expense) => {
        //only if startdate is a number do we want to use it
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day'): true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day'): true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        //figure out if expenses description has the text variable string inside of it
        //includes
        //convert both strings to lowercase
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            //if a is true then 1 else -1
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy ==='amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
};