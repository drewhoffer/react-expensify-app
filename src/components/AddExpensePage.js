import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';

export class AddExpensePage extends React.Component {
    onSubmit=(expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        onSubmit={this.onSubmit}
                    />   
                    <Link className="button button--secondary" to="/">Cancel</Link>
                </div>
   
            </div>
        )
    }
}



const mapDispatchToProps = (dispatch) => ({ 
    startAddExpense: (expense) => dispatch(startAddExpense(expense)) 
});


export default connect(undefined, mapDispatchToProps)(AddExpensePage);