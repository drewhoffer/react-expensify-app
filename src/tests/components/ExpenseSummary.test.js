import { shallow } from 'enzyme';
import React from 'react';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';


test('Should display ExpenseSummary correctly with 0 expenses', () => {
    const wrapper = shallow(<ExpenseSummary />);
    
    expect(wrapper).toMatchSnapshot();
});


test('Should display ExpenseSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should display ExpenseSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={3} expensesTotal={235}/>);
    
    expect(wrapper).toMatchSnapshot();
});