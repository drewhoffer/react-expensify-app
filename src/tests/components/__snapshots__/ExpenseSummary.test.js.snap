// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display ExpenseSummary correctly with 0 expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should display ExpenseSummary correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $2.35
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should display ExpenseSummary correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $2.35
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
